spec.html

```diff
@@ -25194,39 +25210,43 @@ <h1>
         1. Set the running execution context's LexicalEnvironment to _env_.
         1. Let _instanceExtraInitializers_ be a new empty List.
         1. Let _staticExtraInitializers_ be a new empty List.
+        1. Let _metadataParent_ be ? Get(_constructorParent_, @@metadata).
```

Comment:
So `_constructorParent_` gets defaulted to %Function.prototype% when there is no `extends` clause. Which means someone can assign to `Function.prototype[Symbol.metadata]` to be the default parent for metadata. That seems... not good? Should we do this Get only if |ClassHeritage?| is present?

Reply:
I like making the `Get` conditional. Alternatively, we could make `Function.prototype[Symbol.metadata]` nonconfigurable-nonwritable `null`.

Reply:
Hmm, one thing I'm realizing here is that all classes get `Symbol.metadata`, I wonder if we should add the metadata object in cases where decorators are applied, so that it's more performant? If so, then nonconfigurable-nonwritable `null` would make more sense, otherwise we could do the `Get` conditional.

Reply:
My intuition is that classes without any decorators don't have a metadata object. But we should probably talk about this in committee?

Reply:
Yeah that was my original intention, but I just realized on reading that I hadn't made that part optional in the spec. We can discuss in committee.